py/pysparkling/initializer.py

```diff
@@ -46,21 +45,13 @@ def __add_sparkling_jar_to_spark(sc):
             sw_jar_file = '/' + sw_jar_file
 
         url = jvm.java.net.URL("file://{0}".format(sw_jar_file))
-        # Assuming that context class loader is always instance of URLClassLoader
-        # ( which should be always true)
-        cl = jvm.Thread.currentThread().getContextClassLoader()
 
-        # Explicitly check if we run on databricks cloud since there we must add the jar to
-        # the parent of context class loader
-        if cl.getClass().getName() == 'com.databricks.backend.daemon.driver.DriverLocal$DriverLocalClassLoader':
-            cl.getParent().getParent().addURL(url)
-        else:
-            spark_cl = Initializer.__find_spark_cl(cl, 'org.apache.spark.util.MutableURLClassLoader')
-            spark_cl.addURL(url)
+        loader = Initializer.__find_spark_cl(jvm)
```

Comment:
@jakubhava WDYT about adding the url into each class loader from the hierarchy that has the `addUrl` method. I'm just thinking about cases where Spark is breaking standard lookup from parents to children. E.g. ```ExecutorClassLoader```.

Reply:
Good idea! Will do

Reply:
Just one more note to that: I would also add some logging... (the list of loaders where url was added or log an error if there isn't such a classloader)